### PR TITLE
Enable f sharp70

### PR DESCRIPTION
**Pull request by KevinRansom**
- Update F# Language version to 7.0
- Update FSharp.Core to 7.0.0.0
- Update FSharp.Core.nuget to 7.0.0
- Update FCS Compiler Package to 42.7.100.0  - aligned with dotnet 7.0.100   sdk
- Fix fsharp.core assembly version on proto build to match version shipped with vs, remove nuget package hack
- Fix tests that use preview to use LangVersion70.

### CODE REVIEW

**Comment by abelbraaksma** [comment]
Is this per chance the reason why local `build.cmd` is currently failing on constructs with no space between the `<` and `^t`? I know that was part of #13119, but after I updated my fork I got a couple dozen errors, even after cleaning and running `build.cmd` again.

If not, I’ll continue research why this happens, or report separately.

**Comment by vzarytovskii** [comment]
> Is this per chance the reason why local `build.cmd` is currently failing on constructs with no space between the `<` and `^t`? I know that was part of #13119, but after I updated my fork I got a couple dozen errors, even after cleaning and running `build.cmd` again.
> 
> If not, I’ll continue research why this happens, or report separately.

It sounds like proto compiler not being rebuild, and older one is used.

**Comment by abelbraaksma** [comment]
> It sounds like proto compiler not being rebuild, and older one is used.

@vzarytovskii, that was my initial thought. I’ll doublecheck if all artefacts get properly removed with `git clean -xdf` and try again. But from the log, I did see it rebuild. Perhaps the file was in use and I missed that. 🤔

**Comment by abelbraaksma** [comment]
@vzarytovskii: sorry for the noise. Somehow I had more recent `Fharp.Core.Test` files, including the `<^t>` syntax,, but the older compiler. A proper `git fetch upstream` etc solved it.

File: tests/fsharp/typecheck/sigs/temp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+
```
Comment (vzarytovskii) [inline tests/fsharp/typecheck/sigs/temp:1]: Are those here on purpose, in the temp directory?

Comment (KevinRansom) [reply tests/fsharp/typecheck/sigs/temp]: removed

File: src/Compiler/Facilities/LanguageFeatures.fs [hunk 2]
```diff
@@ -110,17 +111,19 @@ type LanguageVersion(versionText) =
...
+                LanguageFeature.ReallyLongLists, languageVersion70
+                LanguageFeature.ErrorOnDeprecatedRequireQualifiedAccess, languageVersion70
+                LanguageFeature.RequiredPropertiesSupport, languageVersion70
+                LanguageFeature.InitPropertiesSupport, languageVersion70
+                LanguageFeature.LowercaseDUWhenRequireQualifiedAccess, languageVersion70
+                LanguageFeature.InterfacesWithAbstractStaticMembers, languageVersion70
+                LanguageFeature.SelfTypeConstraints, languageVersion70
+
                 // F# preview
-                LanguageFeature.FromEndSlicing, previewVersion
```
[the remark this answers is not in the document]
Comment (KevinRansom) [reply src/Compiler/Facilities/LanguageFeatures.fs:114]: removed

File: src/FSharp.Core/prim-types.fsi [hunk 3]
```diff
@@ -2555,7 +2555,6 @@ namespace Microsoft.FSharp.Collections
         /// <param name="offset">The offset from the end.</param>
         ///
         /// <returns>The corresponding index from the start.</returns>
-        [<Experimental("Experimental library feature, requires '--langversion:preview'")>]
```
Comment (vzarytovskii) [inline src/FSharp.Core/prim-types.fsi:2558]: Since the from end slicing remains in preview, shall those methods too?

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**